ReactAndroid/src/main/java/com/facebook/react/modules/websocket/WebSocketModule.java

```diff
@@ -86,6 +87,21 @@ public void connect(final String url, @Nullable final ReadableArray protocols, @
       }
     }
 
+    if (protocols != null && protocols.size() > 0) {
+      StringBuilder protocolsValue = new StringBuilder("");
+      for (int i = 0; i < protocols.size(); i++) {
+        String v = protocols.getString(i).trim();
+        if (!TextUtils.isEmpty(v) && !v.contains(",")) {
+          protocolsValue.append(v);
+          protocolsValue.append(",");
+        }
+      }
+      if (!TextUtils.isEmpty(protocolsValue)) {
+        protocolsValue.replace(protocolsValue.length() - 1, protocolsValue.length(), "");
```

Comment:
well, I'm not familiar with java though, I get `error: cannot find symbol` when using `String.join()` instead `TextUtils.join()`.

**reason 1**
I guess it is because `String.join()` doesn't exist on current android api spec. ( It only appears in java8 ), so it should use TextUtils if you want a join function.
[String API](http://developer.android.com/reference/java/lang/String.html)
[TextUtils API](http://developer.android.com/reference/android/text/TextUtils.html)

**reason 2**
TextUtil.join() requires `Iterable` or `Object[]` instead `ReadableArray`.
so I got error:
```
method TextUtils.join(CharSequence,Iterable) is not applicable
(argument mismatch; ReadableArray cannot be converted to Iterable)
```
Maybe using `StringBuilder` directly is better than convert `ReadableArray` to `Iterable`? 

**reason 3**
I referenced similar implementation from [camera module](https://github.com/facebook/react-native/blob/master/ReactAndroid/src/main/java/com/facebook/react/modules/camera/CameraRollManager.java#L283-L288), which used to build `comma-split SQL IN statement`